src/Features/CSharp/Portable/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -40,79 +39,75 @@ internal class CSharpCodeCleanupService : ICodeCleanupService
-
-                (CSharpFeaturesResources.Add_accessibility_modifiers,
-                 CodeCleanupOptions.AddAccessibilityModifiers,
-                 ImmutableArray.Create(IDEDiagnosticIds.AddAccessibilityModifiersDiagnosticId)),
-
-                (CSharpFeaturesResources.Sort_accessibility_modifiers,
-                 CodeCleanupOptions.SortAccessibilityModifiers,
-                 ImmutableArray.Create(IDEDiagnosticIds.OrderModifiersDiagnosticId)),
-
-                (CSharpFeaturesResources.Make_private_field_readonly_when_possible,
-                 CodeCleanupOptions.MakePrivateFieldReadonlyWhenPossible,
-                 ImmutableArray.Create(IDEDiagnosticIds.MakeFieldReadonlyDiagnosticId)),
-
-                (CSharpFeaturesResources.Remove_unnecessary_casts,
-                 CodeCleanupOptions.RemoveUnnecessaryCasts,
-                 ImmutableArray.Create(IDEDiagnosticIds.RemoveUnnecessaryCastDiagnosticId)),
-
-                (CSharpFeaturesResources.Apply_expression_block_body_preferences,
-                 CodeCleanupOptions.ApplyExpressionBlockBodyPreferences,
-                 ImmutableArray.Create(IDEDiagnosticIds.UseExpressionBodyForConstructorsDiagnosticId,
-                                       IDEDiagnosticIds.UseExpressionBodyForMethodsDiagnosticId,
-                                       IDEDiagnosticIds.UseExpressionBodyForConversionOperatorsDiagnosticId,
-                                       IDEDiagnosticIds.UseExpressionBodyForOperatorsDiagnosticId,
-                                       IDEDiagnosticIds.UseExpressionBodyForPropertiesDiagnosticId,
-                                       IDEDiagnosticIds.UseExpressionBodyForIndexersDiagnosticId,
-                                       IDEDiagnosticIds.UseExpressionBodyForAccessorsDiagnosticId)),
-
-                (CSharpFeaturesResources.Apply_inline_out_variable_preferences,
-                 CodeCleanupOptions.ApplyInlineOutVariablePreferences,
-                 ImmutableArray.Create(IDEDiagnosticIds.InlineDeclarationDiagnosticId)),
-
-                (CSharpFeaturesResources.Remove_unused_variables,
-                 CodeCleanupOptions.RemoveUnusedVariables,
-                 ImmutableArray.Create(CSharpRemoveUnusedVariableCodeFixProvider.CS0168,
-                                       CSharpRemoveUnusedVariableCodeFixProvider.CS0219)),
-
-                (CSharpFeaturesResources.Apply_object_collection_initialization_preferences,
-                 CodeCleanupOptions.ApplyObjectCollectionInitializationPreferences,
-                 ImmutableArray.Create(IDEDiagnosticIds.UseObjectInitializerDiagnosticId,
-                                       IDEDiagnosticIds.UseCollectionInitializerDiagnosticId))
+                (new DiagnosticSet(CSharpFeaturesResources.Apply_implicit_explicit_type_preferences,
+                    new[] { IDEDiagnosticIds.UseImplicitTypeDiagnosticId, IDEDiagnosticIds.UseExplicitTypeDiagnosticId }),
```

Comment:
paramarrays would make ths cleaner.

Reply:
➡️ Consolidating this with https://github.com/dotnet/roslyn/pull/30039#discussion_r226557628